### PR TITLE
New map: Aircraft carrier

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary

Content "New map: Aircraft carrier"

#### Purpose of change

Adding a new aircraft carrier map in game

#### Describe the solution

A full size map that tries to recreate the appearance of aircraft carrier - mostly relied by CV-60 Saratoga blueprints (decommissioned at 94) and some information about HMS Queen Elizabeth

#### Also adds

furniture and terrain:
Triple berth (aka triple bunk bed)
computers, that stay on a wooden desk (in the future ill change them to metal one, and change the computers itself)
locked interior metal doors
elevator without roof (technical thingy)
some deck coating
nuclear reactor (same as aftershock, has no usability for now)

bunch of loot groups

#### Screenshots
screenshots is a bit outdated, but the biggest part of the map is almost unchanged
<details>

floor -1 (reactor room) (this giant spherical thingy is how i tried to make a reactor, in the very middle must be reactors instead of ~, but now there is a dummie terrain)
image
floor 0 (mostly berths)
image
floor 1 (the hangar)
image
floor 2 (flight deck)
image
floor 3 (bridge/air control room)
image

</details>

#### Future
i tried to add some scenario, so testing would be easier, but in some reason the game cannot find any carriers around (they rare it is, but not as rare, alcatraz can be found after 1-2 search attempt, and carrier start cannot spawn me even after 20-30 attemprt)
There is some space for aircraft thingy like planes, armament, aerial bombs etc
need to improve the doors in the future - the one in the walkways must be unpickable
remake the firefight and first aid cabinets (maybe by using collection)

### CODE REVIEW

**Comment by SariusSkelrets** [comment]
The idea is nice but for the story something other than long-ranged teleportation would make more sense if we place it on water. Such a circumstance would be appropriate if you wanted to make it spawn in the middle of a road or something

The story could be about an aircraft carrier that was deployed in or near new england as the riots were going out of control to give support for the ground operations but fell victim to the violence as people were turning feral on the ship.

Theres a lot of options to justify why the boat cannot move anymore: A feral might've sabotaged the reator so it's unsafe to use, the rotors might've been torn apart by something, the Aircraft carrier could be half on the ground and is now unable to move, ect

Since there's a minimal depth required to move an aircraft carrier, I suggest that it only spawns in big bodies of water unless beaching it so the player knows why the boat won't move

Also, the fact you added a story means the boat will need a UNIQUE or GLOBALLY_UNIQUE flag to prevent multiple spawns

For the entity, I don't know exactly about how much anomalous something must be to be outright rejected,
 but I'm pretty sure an entity with mass teleportation abilities, a desire to kill as many people it can, an intelligence at least similar to those of a human and that for some reason stopped all activity after a killing sprew won't be added in the base game

If you want to keep it, you could introduce it as a hallucination of the captain that told him to do something for some reason, like destroying the reactor to save his family.

It illustrates his mental problems and allow you to explain why he did that without bringing in anomalous entities that will probably won't fit the game

Lastly, when writting a lot of text, adding line breaks does wonders to make it easy to read.

**Comment by GuardianDll** [comment]
> The story could be about an aircraft carrier that was deployed in or near new england
thats the problem, the game has no coast line at this moment, and all waterways to inside lakes are too small for such big ship (we had a discussion about it, and the biggest way is St. Lawrence Seaway which is limited to 24 meters in width, when carrier is almost 72 meters)

> Theres a lot of options to justify why the boat cannot move anymore
True, but i write it in such way not to justify "why reactor are gone" but as tool to justify the teleportation thingy

> Also, the fact you added a story means the boat will need a UNIQUE or GLOBALLY_UNIQUE flag to prevent multiple spawns

yeah, its pretty big and full of good loot, so it cant be too frequent location (also there is not so much carriers in the whole world and especially in America)

>For the entity

maybe

> Lastly

my bad

**Comment by SariusSkelrets** [comment]
> thats the problem, the game has no coast line at this moment, and all waterways to inside lakes are too small for such big ship (we had a discussion about it, and the biggest way is St. Lawrence Seaway which is limited to 24 meters in width, when carrier is almost 72 meters)

> True, but i write it in such way not to justify "why reactor are gone" but as tool to justify the teleportation thingy

Then I suppose the captain could've ran the carrier into a tear in reality and ended up in the middle of nowhere, like the player can do ingame by walking into tears but at a much larger scale and distance.

 It also brings the advantage of not caring about water as it entered through a portal and got out by another

**Comment by carmakazi** [comment]
I feel like something massive like this would benefit from the game being able to load and store a separate, contained mini-overmap.  That way it could be "20 miles off the coast adrift" and not interfere with the normal map chunks.  You'd instead get there with a sort of fast-travel prompt with a seaworthy boat or airworthy helicopter (and being able to fly it), or a faction providing passage with one of those things (Old Guard?).

That is core, back end work, though, and I'm nowhere near qualified to work on that.  I do feel like it would be worth it because it has potential to make other story ideas and game functions possible.

I don't know how I feel about dropping this thing in the middle of a normal overmap, but I love the map itself.  I've thought about something like this in the game for a while and it's exactly the kind of scale I was imagining.  I hope you can figure out some way of integrating it into the game.

**Comment by GuardianDll** [comment]
Integrating such stuff was a problem before, when i just start, but with all this portal thingy its not an issue anymore
Also, such big stuff standing in the middle of the lake would feel like nothing in fact - the lakes are still pretty big

**Comment by Wyrdean** [comment]
As far as item four, it makes sense for it to have a lot of loot, seeing as it's unlikely to have been looted previously.

**Comment by GuardianDll** [comment]
Its hard to tell where the problem is, but i assure you there is something really wrong here
just test it by yourself, and you will probably get it

**Comment by Wyrdean** [comment]
> Its hard to tell where the problem is, but i assure you there is something really wrong here just test it by yourself, and you will probably get it

Just to clarify, are you asking me to test the aircraft carrier and try to debug it?

**Comment by GuardianDll** [comment]
If you interested enough, I'll be appreciated

File: data/json/overmap/overmap_special/aircraft_carrier.json [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+      { "point": [ 0, 10, 0 ], "overmap": "aircraft_carrier_11a_north" },
+      { "point": [ 1, 10, 0 ], "overmap": "aircraft_carrier_11b_north" },
+      { "point": [ 2, 10, 0 ], "overmap": "aircraft_carrier_11c_north" },
+      { "point": [ 0, 11, 0 ], "overmap": "aircraft_carrier_12a_north" },
+      { "point": [ 1, 11, 0 ], "overmap": "aircraft_carrier_12b_north" },
+      { "point": [ 2, 11, 0 ], "overmap": "aircraft_carrier_12c_north" },
+      { "point": [ 0, 12, 0 ], "overmap": "aircraft_carrier_13a_north" },
+      { "point": [ 1, 12, 0 ], "overmap": "aircraft_carrier_13b_north" },
+      { "point": [ 2, 12, 0 ], "overmap": "aircraft_carrier_13c_north" },
+      { "point": [ 0, 0, -1 ], "overmap": "aircraft_carrier_1a_-1_north" },
```
Comment (Night-Pryanik) [inline data/json/overmap/overmap_special/aircraft_carrier.json:45]:
```suggestion
      { "point": [ 0, 0, -1 ], "overmap": "aircraft_carrier_1a_-1_north", "locations": [ "lake_water_cube" ] },
```

Add `"locations": [ "lake_water_cube" ]` to all underwater parts of your carrier.

**Comment by Night-Pryanik** [comment]
I strongly recommend you to surround your carrier with a minimum of 2-tile-wide belt of water, like I did for island prisons. Check `Alcatraz Prison` overmap special on how to do it. It will create a more "natural" look, otherwise the could spawn your carrier right next to the shore, and it might look "bad". Unless you want to spawn your carrier right next to the shore, of course.

File: data/json/overmap/overmap_special/aircraft_carrier.json (hunk 1 above)
Comment (GuardianDll) [reply data/json/overmap/overmap_special/aircraft_carrier.json:45]: God, i dont know what type of magic do you use, but it fixed my spawn problem
thank you

**Pull request merged**